webcurator-core/src/main/java/org/webcurator/core/store/PywbWarcDeposit.java

```diff
@@ -0,0 +1,253 @@
+            throw new DigitalAssetStoreException(err);
+        }
+
+        String wb_manager = ProcessBuilderUtils.getFullPathOfCommand("wb-manager");
+        if (wb_manager == null) {
+            log.error("The command tool [wb-manager] is not installed or not in the env PATH.");
+            return;
+        }
+
+        //Added all warc files to the PYWB
+        for (File warc : warcFiles) {
+            String[] commands = {wb_manager, "add", pywbManagerColl, warc.getAbsolutePath()};
+            List<String> commandList = Arrays.asList(commands);
+            ProcessBuilder processBuilder = new ProcessBuilder(commandList);
+            processBuilder.directory(pywbManagerStoreDir);
+            try {
+                Process process = processBuilder.inheritIO().start();
+                int processStatus = process.waitFor();
+
+                if (processStatus != 0) {
+                    throw new Exception("Process ended with a failed status: " + processStatus);
+                }
+            } catch (Exception e) {
+                log.error("Unable to process the command in a new thread.", e);
+                throw e;
+            }
+        }
+    }
+
+    public List<SeedHistoryDTO> getSeedWithTimestamps(HarvestResultDTO harvestResult) throws DigitalAssetStoreException {
+        if (!this.pywbEnabled) {
+            return null;
+        }
+
+        File directory = new File(rootStorePath, harvestResult.getTargetInstanceOid() + File.separator + harvestResult.getHarvestNumber());
+        File[] warcFiles = directory.listFiles((dir, name) -> name.toLowerCase().endsWith(".warc") ||
+                name.toLowerCase().endsWith(".warc.gz"));
+        if (warcFiles == null || warcFiles.length == 0) {
+            String err = "No warc files found in folder: " + directory.getAbsolutePath();
+            log.error(err);
+            throw new DigitalAssetStoreException(err);
+        }
+
+        Map<String, File> mappedWarcFiles = new HashMap<>();
+
+        //Added all warc files to the PYWB
+        for (File warc : warcFiles) {
+            mappedWarcFiles.put(warc.getName(), warc);
+        }
+
+        File[] cdxFiles = directory.listFiles((dir, name) -> name.toLowerCase().endsWith(".cdx"));
+        if (cdxFiles == null) {
+            return null;
+        }
+
+        LocalDateTime minTimestamp = LocalDateTime.now(), maxTimestamp = LocalDateTime.now().minusYears(5L);
+        for (File cdx : cdxFiles) {
+            List<String> lines;
+            try {
+                lines = FileUtils.readLines(cdx);
+            } catch (IOException e) {
+                throw new DigitalAssetStoreException(e);
+            }
+            for (int idx = 1; idx < lines.size(); idx++) {
+                String line = lines.get(idx);
+                int pos = line.indexOf(' ');
+                if (pos <= 1) {
+                    continue;
+                }
+
+                String strTimestamp = line.substring(pos + 1, pos + 15);
```

Comment:
This relies on a fixed format of the CDX file generated by WCT, but 3.1.3 introduced configurable CDX formats, so this will not work for users who don't have a timestamp in the second field of their CDX files.

Reply:
I've reverted to a fixed CDX format in PR #92